### ISSUE TITLE
Exception when comparing List<Dictionary<string, string>>

### ISSUE THREAD

**Issue by carsten-j** [GregFinzer/Compare-Net-Objects#38]
When I run the following code

```csharp
var bar1 = new List<Dictionary<string, string>>
{
    new Dictionary<string, string>
    {
        {"a", "b"},
        {"c", "d"},
        {"e", "f"},
        {"g", "h"},
    }
};

var bar2 = new List<Dictionary<string, string>>
{
    new Dictionary<string, string>
    {
        {"e", "f"},
        {"g", "h"},
        {"c", "d"},
        {"a", "b"},
    }
};

var comparer = new CompareLogic {Config = {IgnoreCollectionOrder = true}};
var res = comparer.Compare(bar1, bar2);
```

an exception is thrown:
```
System.Reflection.TargetParameterCountException : Parameter count mismatch.
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture, Boolean skipVisibilityChecks)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, Object[] index)
   at KellermanSoftware.CompareNetObjects.IgnoreOrderTypes.IgnoreOrderLogic.GetMatchIndex(ComparisonResult result, List`1 spec, Object currentObject) in C:\Users\test\Downloads\Compare-Net-Objects-master\Compare-NET-Objects\IgnoreOrderTypes\IgnoreOrderLogic.cs:line 238
   at KellermanSoftware.CompareNetObjects.IgnoreOrderTypes.IgnoreOrderLogic.CompareInOrder(CompareParms parms) in C:\Users\test\Downloads\Compare-Net-Objects-master\Compare-NET-Objects\IgnoreOrderTypes\IgnoreOrderLogic.cs:line 76
   at KellermanSoftware.CompareNetObjects.IgnoreOrderTypes.IgnoreOrderLogic.CompareEnumeratorIgnoreOrder(CompareParms parms, Boolean countsDifferent) in C:\Users\test\Downloads\Compare-Net-Objects-master\Compare-NET-Objects\IgnoreOrderTypes\IgnoreOrderLogic.cs:line 45
   at KellermanSoftware.CompareNetObjects.TypeComparers.ListComparer.CompareType(CompareParms parms) in C:\Users\test\Downloads\Compare-Net-Objects-master\Compare-NET-Objects\TypeComparers\ListComparer.cs:line 57
   at KellermanSoftware.CompareNetObjects.RootComparer.Compare(CompareParms parms) in C:\Users\test\Downloads\Compare-Net-Objects-master\Compare-NET-Objects\RootComparer.cs:line 51
   at KellermanSoftware.CompareNetObjects.CompareLogic.Compare(Object object1, Object object2) in C:\Users\test\Downloads\Compare-Net-Objects-master\Compare-NET-Objects\CompareLogic.cs:line 207
```

Do I need to set additional configuration parameters or it this case not supported?

**Comment by GregFinzer** [comment]
It looks like this is a defect with the IgnoreCollectionOrder.  It will work without ignoring the order:

```
        [Test]
        public void DictionaryWithIgnoreOrder()
        {
            var bar1 = new List<Dictionary<string, string>>
            {
                new Dictionary<string, string>
                {
                    {"a", "b"},
                    {"c", "d"},
                    {"e", "f"},
                    {"g", "h"},
                }
            };

            var bar2 = new List<Dictionary<string, string>>
            {
                new Dictionary<string, string>
                {
                    {"e", "f"},
                    {"g", "h"},
                    {"c", "d"},
                    {"a", "b"},
                }
            };

            var comparer = new CompareLogic { Config = { IgnoreCollectionOrder = false } };
            var res = comparer.Compare(bar1, bar2);
        }
```

It will be a while until I can create a fix for this.

**Comment by carsten-j** [comment]
Thanks for the update. Do you think this will be difficult to fix? I do not know your code base but maybe I could have a look to see if this is something not that difficult to fix for a newcomer to your project!

**Comment by GregFinzer** [comment]
My guess is that it just needs a change in the IgnoreOrderLogic and then not process dictionary types since that is already handled by other code.

**Comment by GregFinzer** [comment]
@carsten-j Fix coming soon

**Comment by GregFinzer** [comment]
Fixed in version 3.08
https://www.nuget.org/packages/CompareNETObjects/3.8.0

**Comment by carsten-j** [comment]
Thanks
